### ISSUE TITLE
Request: Add new formatter - "URL to Image" to image field in Views

### ISSUE THREAD

**Issue by simone960** [backdrop/backdrop-issues#2921]
In Drupal 8, when adding an image field in Views it has an additional "URL to Image" formatter which can output the image url easily. I find this feature is great. Back in Drupal 7 it has also a module called [Image URL Formatter](https://www.drupal.org/project/image_url_formatter) which do the same. I wish this feature can be added to the core. It should be able to to support image style as well. Please consider adding to Views for future enhancement. Thanks!

![url to image2](https://user-images.githubusercontent.com/7530286/34076628-06a187c2-e341-11e7-9e9b-06eae7165ca7.png)

**Comment by olafgrabienski** [comment]
Nice idea! There is another way to get the image URL in Views, but it doesn't work with image styles:

- Add a relationship "File Usage: File".
- Add a field "File: Path", and enable the option "Display download path instead of file storage URI".

**Comment by simone960** [comment]
> Add a relationship "File Usage: File".
> Add a field "File: Path", and enable the option "Display download path instead of file storage URI".

Interesting... I don't know it can be done without a module. But it's pity it doesn't support image styles as you pointed. Most cases I need to use it with the image styles in order to control the output dimension. Anyway, thanks for sharing @olafgrabienski ! I learnt something new today !

**Comment by jenlampton** [comment]
I've wanted this feature often too, and I didn't know about @olafgrabienski's way :)

**Comment by jlfranklin** [comment]
> Back in Drupal 7 it has also a module called Image URL Formatter which do the same.

So, start by porting the module.

**Comment by opi** [comment]
Maybe a simple formatter like this could be in core ?

**Comment by jlfranklin** [comment]
We have a small team that does core reviews and commits.  I'd rather they focus their efforts on core bug fixes and new features that *must* be in core for technical reasons, and leave things that already are D7 modules as Backdrop modules.

By porting the module, @simone960 doesn't need to wait for the core review process, he gets experience developing for Backdrop, Backdrop gets a richer library of contributed modules, and the community expands by +1 contrib developer.  It's wins all around!

When Backdrop 2.0 rolls around, I have no doubt the most popular modules will be candidates for inclusion in core.  Until then, it's a couple clicks to get the Project Browser to download and install URL Image Formatter.

**Comment by jenlampton** [comment]
@opi I'm with you, I think this should be in core. I don't think it should be top priority, but if a patch was ready we would consider it ;)

**Comment by jlfranklin** [comment]
> but if a patch was ready we could consider it ;)

If you're going to craft a patch, do it soon.  Feature freeze for 1.9 about 10 days away.

**Comment by jenlampton** [comment]
Probably not for 1.9...  that's right around the corner... but maybe later in 1.x. My point was only that his isn't an API break so it doesn't need to wait for 2.0.

**Comment by Graham-72** [comment]
I have just ported 'image_url_formatter' to Backdrop - very little needs changing. I will now have a look at what a patch might involve.

**Comment by jlfranklin** [comment]
> My point was only that his isn't an API break so it doesn't need to wait for 2.0. :)

No argument there.  My "when 2.0 rolls around" is premised on the fact that once something is already in a module, it's a much harder sell to get it into core.  Not impossible, Global Redirect may be merged into core, but harder.

On the topic of timelines, a patch before the end of the year might get it into core in mid-January, if the code is clean and the rivers don't rise.  Otherwise, it'll have to wait until April.  OTOH, port the module and it's available today.  Just sayin.'  :)

**Comment by jenlampton** [comment]
> On the topic of timelines, a patch before the end of the year might get it into core in mid-January, if the code is clean and the rivers don't rise. Otherwise, it'll have to wait until April. OTOH, port the module and it's available today. Just sayin.' :)

True, or you could [maintain your own patch](http://www.jenlampton.com/blog/sustainable-development-workflow-patching).

**Comment by Graham-72** [comment]
The contributed module is at https://github.com/backdrop-contrib/image_url_formatter.

Turning this into a PR for core looks to be beyond my capability. Would someone else like to do this?

**Comment by klonos** [comment]
Thanks for porting the module @Graham-72 👍

I'm with @jenlampton and @opi on this. It should be in core.

> Interesting... I don't know it can be done without a module. But it's pity it doesn't support image styles as you pointed. Most cases I need to use it with the image styles in order to control the output dimension.

https://www.drupal.org/project/imagecache_token

> This module provides additional tokens for image fields. For each image style available, a token `[node:field_image_field:style_name]` will be provided. It is also possible to get image properties like width or height by using tokens like `[node:field_image_field:style_name:height]`.

...another one to port I guess 😉
